File: Infrastructure/ZIndexConverter.fs
namespace BattleCity.Infrastructure

open Avalonia.Data.Converters
open BattleCity.Model
open System

type ZIndexConverter internal () =
    interface IValueConverter with
        member _.Convert (value, targetType, parameter, culture) : obj =
            match value with
            | :? Player -> 2
            | :? Tank -> 1
            | _ -> 0
        member _.ConvertBack (value, targetType, parameter, culture) : obj =
            raise <| NotImplementedException()
        
    static member Instance = ZIndexConverter()

File: Infrastructure/ZIndexConverter.cs
namespace BattleCity.Infrastructure;

using Avalonia.Data.Converters;
using Model;
using System;
using System.Globalization;

internal class ZIndexConverter : IValueConverter {
    public static ZIndexConverter Instance { get; } = new();

    public object Convert(object value, Type targetType, object parameter, CultureInfo culture) =>
        value switch
        {
            Player => 2,
            Tank => 1,
            _ => 0
        };

    public object ConvertBack(object value, Type targetType, object parameter, CultureInfo culture) =>
        throw new NotImplementedException();
}